packages/modal-button/src/ModalButton.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ForwardedRef, forwardRef } from 'react'
+
+import {
+  ModalButtonStyle,
+  ModalButtonContentStyle,
+  // ModalButtonContentSvgStyle,
+} from './ModalButtonStyles'
+import { ModalButtonIconProps } from './types'
+
+function ModalButton(
```

Comment:
so, this is a political issue :)

I think that inheritance is bad way. Why? Because we will lost components isolation from each other.

Repeat yourself for components  isolation :).